doc/model.rst

```diff
@@ -0,0 +1,346 @@
+*****
+Model
+*****
+========
+Overview
+========
+Model(s) are subclasses of SmartSimEntity(s) and are created through the
+Experiment API. Models represent any computational kernel. Models are flexible
+enough to support many different applications, however, to be used with our
+clients (SmartRedis) the application will have to be written in Python, C, C++, or Fortran.
+
+Models are given RunSettings objects that specify how a kernel should be executed
```

Comment:
I think we need more overview content.  This doesn't really introduce the key aspects of the ``Model`` object, how it is used, and what information you need to create one.  We should give more general context and then talk about what will be explained in the other sections.  I think there is a general lack of continuity between the overview and subsequent sections.  

A good starting point (but don't copy verbatim) is the doc strings for the Model API: 
https://www.craylabs.org/develop/api/smartsim_api.html#smartsim.experiment.Experiment.create_model